### PR TITLE
Feature/discovery/choose language modal

### PR DESCRIPTION
**Pull request by camillebrulotte**
**#66 is merged in this branch. #66 needs to be merged before.** 
The base branch use is the main_discovery to not show all the duplicate changes. 
But, this branch will ultimately be merged in master.

In the video, after selecting the language, the app is navigating to the schedule page. Normally, it will show the dashboard page.

https://user-images.githubusercontent.com/37625944/111083708-31a1da00-84e5-11eb-9263-1ddd18b43bdc.mp4

image

### CODE REVIEW

File: test/ui/views/choose_language_view_test.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  AppIntl intl;
+  group('SettingsView - ', () {
+    setUp(() async {
+      intl = await setupAppIntl();
+      setupNavigationServiceMock();
+      setupCacheManagerMock();
+      setupSettingsManagerMock();
+    });
+
+    tearDown(() {});
```
Comment (MysticFragilist) [inline test/ui/views/choose_language_view_test.dart:21]: I'm not sure, but I think in the tearDown, you need to call `unregister<OneOfTheServiceYouUsed>();` for each service you have setup (I found an example [here](https://github.com/ApplETS/Notre-Dame/blob/master/test/ui/views/login_view_test.dart))

**Review by MysticFragilist — CHANGES_REQUESTED** [review]

File: lib/core/constants/router_paths.dart [hunk 2]
```diff
@@ -10,4 +10,5 @@ class RouterPaths {
   static const String settings = "/more/settings";
   static const String contributors = "/more/contributors";
   static const String about = "/more/about";
+  static const String chooseLanguage = "/choose-language";
```
Comment (apomalyn) [inline lib/core/constants/router_paths.dart:13]:
```suggestion
  static const String chooseLanguage = "/discovery/choose-language";
```

**Review by apomalyn — CHANGES_REQUESTED** [review]
Because this would probably be the first screen you see after your first login I think a popup isn't the best way to present this information. 

What do you think about a screen instead of a modal? After this screen, we could be redirected to a screen explaining what is this ETSMobile, ... what do you think about that? If needed we could discuss this during the next meeting.

**Comment by camillebrulotte** [comment]
> Because this would probably be the first screen you see after your first login I think a popup isn't the best way to present this information. 
> 
> 
> 
> What do you think about a screen instead of a modal? After this screen, we could be redirected to a screen explaining what is this ETSMobile, ... what do you think about that? If needed we could discuss this during the next meeting.

2 weeks ago we discussed about it and stumbled upon the modal as a solution since the "full page" was reserved for "what's new". I think that's what I understood ? @AlexandreMarkus @emondhar @MysticFragilist 

But when I think about it again, a full page may be a better idea than the modal.

**Comment by AlexandreMarkus** [comment]
Same as @camillebrulotte , but I think we should discuss it again

**Comment by camillebrulotte** [comment]
Here are some ideas I came up with : 

[Choose_language.pdf](https://github.com/ApplETS/Notre-Dame/files/6185794/Choose_language.pdf)

Feel free if you have a better idea :D

**Comment by apomalyn** [comment]
@camillebrulotte first of all, that was QUICK! :smile: 
I really like option A but there are some issues:

1. We can't use the French and UK flag because we are in Canada (and so use French Canadian and English Canadian), so we could remove the flags and turn the tiles in button (with card)
2. The light version of the screen should have a red background and a white logo to follow the theme of the previous and next screen.

What do you think ? :smile: (and by the way, great job :tada: )

**Comment by AlexandreMarkus** [comment]
Same as @apomalyn I like option A

**Review by MysticFragilist — DISMISSED** [review]

**Comment by MysticFragilist** [comment]
Oups, just wanted to approve the changes I had prompted before. I didn't know it would approve the PR as a whole 😅

**Comment by camillebrulotte** [comment]
Parff, option A it iss

**Comment by camillebrulotte** [comment]
Voilà ! :D 

https://user-images.githubusercontent.com/37625944/112700903-92c29980-8e65-11eb-9658-17280b2bd110.mp4

https://user-images.githubusercontent.com/37625944/112700906-935b3000-8e65-11eb-8644-76001cd2b5ae.mp4

File: lib/core/viewmodels/startup_viewmodel.dart [hunk 3]
```diff
@@ -5,11 +5,20 @@ import 'package:stacked/stacked.dart';
 import 'package:notredame/core/managers/user_repository.dart';
 import 'package:notredame/core/services/navigation_service.dart';
 
+// MANAGER
+import 'package:notredame/core/managers/settings_manager.dart';
```
Comment (apomalyn) [inline lib/core/viewmodels/startup_viewmodel.dart:9]: Maybe we could merge these two sections? :thinking:

**Review by apomalyn — CHANGES_REQUESTED** [review]

**Review by apomalyn — APPROVED** [review]
LGTM :1st_place_medal: :tada:

**Pull request merged**